pkg/rcmanager/subnet.go

```diff
@@ -0,0 +1,316 @@
+			klog.Error(err.Error())
+			m.Recorder.Event(subnet, v1.EventTypeWarning, ReasonSubnetConflict, err.Error())
+			return errors.Newf("Overlap network. overlap with other remoteSubnet")
+		}
+	}
+	return nil
+}
+
+// Reconcile local cluster *networkingv1.RemoteSubnet based on remote cluster's subnet.
+func (m *Manager) diffSubnetAndRCSubnet(subnets []*networkingv1.Subnet, rcSubnets []*networkingv1.RemoteSubnet,
+	networkMap map[string]*networkingv1.Network) (
+	add []*networkingv1.Subnet, update []*networkingv1.RemoteSubnet, remove []*networkingv1.RemoteSubnet) {
+	subnetMap := func() map[string]*networkingv1.Subnet {
+		subnetMap := make(map[string]*networkingv1.Subnet)
+		for _, s := range subnets {
+			subnetMap[s.Name] = s
+		}
+		return subnetMap
+	}()
+
+	for _, v := range rcSubnets {
+		if v.ClusterName != m.ClusterName {
+			continue
+		}
+		if subnet, exists := subnetMap[v.Name]; !exists {
+			remove = append(remove, v)
+		} else {
+			newestRemoteSubnet, err := m.convertSubnet2RemoteSubnet(subnet, networkMap[subnet.Spec.Network])
+			if err != nil {
+				continue
+			}
+			if !reflect.DeepEqual(newestRemoteSubnet.Spec, v.Spec) {
+				update = append(update, newestRemoteSubnet)
+			}
+		}
+	}
+	remoteSubnetMap := func() map[string]*networkingv1.RemoteSubnet {
+		remoteSubnetMap := make(map[string]*networkingv1.RemoteSubnet)
+		for _, s := range rcSubnets {
+			remoteSubnetMap[s.Name] = s
+		}
+		return remoteSubnetMap
+	}()
+
+	for _, s := range subnets {
+		remoteSubnetName := utils.GenRemoteSubnetName(m.ClusterName, s.Name)
+		if _, exists := remoteSubnetMap[remoteSubnetName]; !exists {
+			add = append(add, s)
+		}
+	}
+	return
+}
+
+func (m *Manager) convertSubnet2RemoteSubnet(subnet *networkingv1.Subnet, network *networkingv1.Network) (*networkingv1.RemoteSubnet, error) {
+	if network == nil {
+		return nil, errors.Newf("Subnet corresponding Network is nil. Subnet=%v, Cluster=%v", subnet.Name, m.ClusterName)
+	}
+	rs := &networkingv1.RemoteSubnet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: utils.GenRemoteSubnetName(m.ClusterName, subnet.Name),
+			Labels: map[string]string{
+				constants.LabelCluster: m.ClusterName,
+				constants.LabelSubnet:  subnet.Name,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: networkingv1.SchemeGroupVersion.String(),
+					Kind:       "RemoteCluster",
+					Name:       m.ClusterName,
+					UID:        m.RemoteClusterUID,
+					Controller: pointer.BoolPtr(true),
+				},
+			},
+		},
+		Spec: networkingv1.RemoteSubnetSpec{
+			Range:       subnet.Spec.Range,
+			Type:        network.Spec.Type,
+			ClusterName: m.ClusterName,
+		},
+	}
+	return rs, nil
+}
+
+func (m *Manager) filterSubnet(obj interface{}) bool {
+	if !m.GetIsReady() {
+		return false
+	}
+	_, ok := obj.(*networkingv1.Subnet)
+	return ok
+}
+
+func (m *Manager) addOrDelSubnet(obj interface{}) {
+	subnet, _ := obj.(*networkingv1.Subnet)
+	m.EnqueueSubnet(subnet.ObjectMeta.Name)
```

Comment:
If `subnet.ObjectMeta.Name` is not used, maybe use `rcmanager.ReconcileSubnet` is better.